compiler/rustc_trait_selection/src/traits/wf.rs

```diff
@@ -970,10 +970,10 @@ pub fn object_region_bounds<'tcx>(
     existential_predicates: &'tcx ty::List<ty::PolyExistentialPredicate<'tcx>>,
 ) -> Vec<ty::Region<'tcx>> {
     let predicates = existential_predicates.iter().filter_map(|predicate| {
-        if let ty::ExistentialPredicate::Projection(_) = predicate.skip_binder() {
-            None
-        } else {
+        if let ty::ExistentialPredicate::Trait(_) = predicate.skip_binder() {
             Some(predicate.with_self_ty(tcx, tcx.types.trait_object_dummy_self))
+        } else {
+            None
```

Comment:
this relies on auto-traits not having any region bounds :thinking: I don't like the fact that we implicitly rely on that here, given that we may change it in the future, even if I can't think of any auto trait which does so.

Why don't we just pass in all `ExistentialPredicates`  here without any filtering? perf? the elaborator should filter duplicate predicates